src/lib/DefaultRuleManager.ts

```diff
@@ -75,10 +75,10 @@ export class DefaultRuleManager implements RuleManager {
 		const executedEngines: Set<string> = new Set();
 
 		// Derives rules from our filters to feed the engines.
-		const ruleGroups: RuleGroup[] = this.catalog.getRuleGroupsMatchingFilters(filters);
+		const engines: RuleEngine[] = await this.resolveEngineFilters(filters, engineOptions, runOptions);
```

Comment:
Engines are now filtered before rule groups, and provided as input to the rule group filtering process.